lib/ex_webrtc/peer_connection.ex

```diff
@@ -717,67 +722,58 @@ defmodule ExWebRTC.PeerConnection do
 
   @impl true
   def handle_call({:set_transceiver_direction, tr_id, direction}, _from, state) do
-    idx = Enum.find_index(state.transceivers, fn tr -> tr.id == tr_id end)
-
-    case idx do
-      nil ->
-        {:reply, {:error, :invalid_transceiver_id}, state}
-
-      idx ->
-        tr = Enum.at(state.transceivers, idx)
-        tr = %{tr | direction: direction}
+    state.transceivers
+    |> Enum.with_index()
+    |> Enum.find(fn {tr, _idx} -> tr.id == tr_id end)
+    |> case do
+      {tr, idx} ->
+        tr = RTPTransceiver.set_direction(tr, direction)
         transceivers = List.replace_at(state.transceivers, idx, tr)
         state = %{state | transceivers: transceivers}
         state = update_negotiation_needed(state)
         {:reply, :ok, state}
+
+      nil ->
+        {:reply, {:error, :invalid_transceiver_id}, state}
     end
   end
 
   @impl true
   def handle_call({:stop_transceiver, tr_id}, _from, state) do
-    idx = Enum.find_index(state.transceivers, fn tr -> tr.id == tr_id end)
+    state.transceivers
+    |> Enum.with_index()
+    |> Enum.find(fn {tr, _idx} -> tr.id == tr_id end)
+    |> case do
```

Comment:
While those changes are okay I feel like this is only refactoring? If yes, in general, I would put such changes in a separate PR :wink:

Reply:
sure, I came to the same conclusion, but it's not much so I left it here